tls/s2n_kem_preferences.c

```diff
@@ -30,16 +50,24 @@ const struct s2n_kem_group *pq_kem_groups_r3[] = {
 const struct s2n_kem_preferences kem_preferences_pq_tls_1_0_2021_05 = {
     .kem_count = s2n_array_len(pq_kems_r3_2021_05),
     .kems = pq_kems_r3_2021_05,
-    .tls13_kem_group_count = s2n_array_len(pq_kem_groups_r3),
-    .tls13_kem_groups = pq_kem_groups_r3,
+    .tls13_kem_group_count = s2n_array_len(pq_kem_groups_r3_2021_05),
+    .tls13_kem_groups = pq_kem_groups_r3_2021_05,
     .tls13_pq_hybrid_draft_revision = 0
 };
 
 const struct s2n_kem_preferences kem_preferences_pq_tls_1_0_2023_01 = {
     .kem_count = s2n_array_len(pq_kems_r3_2021_05),
     .kems = pq_kems_r3_2021_05,
-    .tls13_kem_group_count = s2n_array_len(pq_kem_groups_r3),
-    .tls13_kem_groups = pq_kem_groups_r3,
+    .tls13_kem_group_count = s2n_array_len(pq_kem_groups_r3_2021_05),
+    .tls13_kem_groups = pq_kem_groups_r3_2021_05,
+    .tls13_pq_hybrid_draft_revision = 5
+};
+
+const struct s2n_kem_preferences kem_preferences_pq_tls_1_3_2023_06 = {
+    .kem_count = s2n_array_len(pq_kems_r3_2023_06),
+    .kems = pq_kems_r3_2023_06,
```

Comment:
`.kems` is the TLS 1.2 Hybrid PQ mechanism, and `.tls13_kem_groups` is the TLS 1.3 mechanism. The TLS 1.2 Hybrid PQ draft RFC is no longer being updated, so I think we can make `.kems` set to NULL and `.kem_count` to 0, but leave the other values how you have it.

Reply:
ah! that makes a lot of sense. will do.

Reply:
thinking through this a bit more, if don't support any TLS 1.2 KEMs, won't that silently "downgrade" older PQ TLS clients supporting < 1.3 to classical-only ECDHE connections? if so, is that acceptable?

i ran a quick test with old PQ client + new PQ server, and it appears to negotiate TLS 1.2 without any hybrid KEM/KEM Groups.

client:

```
$ ./build/bin/s2nc --non-blocking -e -T --insecure -c KMS-PQ-TLS-1-0-2020-07 localhost 19013 | grep -E 'Actual protocol|Handshake|KEM'
Handshake: NEGOTIATED|FULL_HANDSHAKE|TLS12_PERFECT_FORWARD_SECRECY
Actual protocol version: 33
KEM: NONE
KEM Group: NONE
```

server:

```
$ ./build/bin/s2nd -X --self-service-blinding --non-blocking --key ./tests/pems/rsa_4096_sha512_client_key.pem --cert ./tests/pems/rsa_4096_sha512_client_cert.pem -t ./tests/pems/rsa_4096_sha512_client_cert.pem -c PQ-TLS-1-3-2023-06-01 -T localhost 19013 | grep -E 'Actual protocol|Handshake|KEM'
Handshake: NEGOTIATED|FULL_HANDSHAKE|TLS12_PERFECT_FORWARD_SECRECY
Actual protocol version: 33
KEM: NONE
KEM Group: NONE
```

Reply:
ok,  after looking more closely at the [which security policy the SDK currently uses](https://github.com/awslabs/aws-c-io/blame/e87baa29fb621c8b42146c0206867f012f1acf23/source/s2n/s2n_tls_channel_handler.c#L1426C69-L1426C90), my prior experiment was used much older, [deprecated](https://github.com/awslabs/aws-c-io/blob/e87baa29fb621c8b42146c0206867f012f1acf23/include/aws/io/tls_channel_handler.h#L36C43-L36C43) policy that's no longer supported. right now, the SDK only supports `PQ-TLS-1-0-2021-05-26`, so that's the "old" client policy we'd need to worry about server-side during deployment of `TLS-1-3-2023-06-01`. I confirmed that this combination ("old" client, new server) does in fact negotiate TLS 1.3 with the hybrid PQ groups.

server:

```
$ ./build/bin/s2nd -X --self-service-blinding --non-blocking --key ./tests/pems/rsa_4096_sha512_client_key.pem --cert ./tests/pems/rsa_4096_sha512_client_cert.pem -t ./tests/pems/rsa_4096_sha512_client_cert.pem -c PQ-TLS-1-3-2023-06-01 -T localhost 19013 | grep -E 'Actual protocol|Handshake|KEM'
Handshake: NEGOTIATED|FULL_HANDSHAKE|MIDDLEBOX_COMPAT
Actual protocol version: 34
KEM: NONE
KEM Group: x25519_kyber-512-r3
```

client:

```
$ ./build/bin/s2nc --non-blocking -e -T --insecure -c PQ-TLS-1-0-2021-05-26 localhost 19013 | grep -E 'Actual protocol|Handshake|KEM'
Handshake: NEGOTIATED|FULL_HANDSHAKE|MIDDLEBOX_COMPAT
Actual protocol version: 34
KEM: NONE
KEM Group: x25519_kyber-512-r3
```